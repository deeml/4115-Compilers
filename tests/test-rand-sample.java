import java.util.*;
import java.lang.*;

class program 
{
//input

//output

private static Random __rand = new Random(System.currentTimeMillis());
private static double random_sample()
{
double u = 0.0;
u = __rand.nextFloat();
if (u < 0.5)
{
return Math.sqrt(0 - 0.5 * 3.141593 * Math.log(1 - Math.pow(2 * u - 1,2)));
}
else
{
return 0 - Math.sqrt(0 - 0.5 * 3.141593 * Math.log(1 - Math.pow(1 - 2 * u,2)));
}
}

public static void main(String[] args){
__rand = new Random(1234);
int i = 0;
i = 0;
double avg = 0.0;
avg = 0.;
while (i < 1000) {
i = i + 1;
avg = avg + random_sample();
}
System.out.println("close to zero");
System.out.println(avg / 1000);

return ;
}
}
