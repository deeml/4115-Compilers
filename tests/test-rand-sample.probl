input{}
output{}
fun random_sample(): float {
	float u;
	u = rand();
	if (u<0.5) {
		return sqrt(-0.5*3.141593*log(1-(2*u-1)^2));
	} else {
		return -sqrt(-0.5*3.141593*log(1-(1-2*u)^2));
	}
}
model
{
	srand(1234);
	int i;
	i=0;
	float avg;
	avg=0.0;
	while (i<1000) {
		avg = avg + random_sample();
		i = i+1;
	}
	print('close to zero');
	print(avg/1000);
}
