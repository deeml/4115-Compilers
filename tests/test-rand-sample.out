close to zero
0.02514514913778494
